src/main/java/de/tum/in/www1/artemis/web/rest/ComplaintResource.java

```diff
@@ -149,10 +155,39 @@ public ComplaintResource(AuthorizationCheckService authCheckService, ExerciseSer
         return ResponseEntity.ok(responseComplaints);
     }
 
+    /**
+     * Get /more-feedback/for-tutor-dashboard/:exerciseId
+     * <p>
+     * Get all the more feedback requests associated to an exercise, that are about the tutor who is doing the request.
+     * @param exerciseId the id of the exercise we are interested in
+     * @return the ResponseEntity with status 200 (OK) and a list of more feedback requests. The list can be empty
+     */
+    @GetMapping("/more-feedback/for-tutor-dashboard/{exerciseId}")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<List<Complaint>> getMoreFeedbackRequestsForTutorDashboard(@PathVariable Long exerciseId, Principal principal) {
+        Exercise exercise = exerciseService.findOne(exerciseId);
+        if (!authCheckService.isAtLeastTeachingAssistantForExercise(exercise)) {
+            return forbidden();
+        }
+
+        List<Complaint> responseComplaints = complaintService.getMyMoreFeedbackRequests(exerciseId, principal);
+        return ResponseEntity.ok(responseComplaints);
+    }
+
+    /**
+     * Get /complaints
+     * <p>
+     * Get all the complaints filtered by different request parameters.
+     * @param tutorId the id of the tutor by which we want to filter
+     * @param exerciseId the id of the exercise we are interested in
+     * @param courseId the id of the course we are interested in
+     * @param complaintType the type of complaints we are interested in
+     * @return the ResponseEntity with status 200 (OK) and a list of complaints. The list can be empty
+     */
     @GetMapping("/complaints")
     @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
-    public ResponseEntity<List<Complaint>> getComplaintsFilteredBy(@RequestParam(required = false) Long tutorId, @RequestParam(required = false) Long exerciseId,
-            @RequestParam(required = false) Long courseId) {
+    public ResponseEntity<List<Complaint>> getComplaintsFilteredBy(@RequestParam() ComplaintType complaintType, @RequestParam(required = false) Long tutorId,
+            @RequestParam(required = false) Long exerciseId, @RequestParam(required = false) Long courseId) {
```

Comment:
This resource method is quite long and basically handles three distinct cases and does not reuse code at all. Therefore, I suggest we split this up into 3 distinct resources that:

1) tutorId --> /tutors/{tutorId}/complaints
2) courseId --> /courses/{courseId}/complaints
3) exerciseId --> /exercise/{exerciseId}/complaints

I would keep all 3 requests in the ComplaintResource for now

Reply:
I checked the usage on the front-end and in this case we would have to add 2 more distinct requests: 
 4. tutorId + courseId
 5. tutorId + exerciseId
I see your point that this method is too complex, but I don't think it makes sense to split it in 5 or 10 (if we distinguish each of them by `complaintType`).

Reply:
No I don't think we need 10 cases.
We can keep complaintType as RequestParam.
In addition 2. and 3. could have an optional RequestParam tutorId.

As we have 3 distinct return statements and even different isAtLeastTutor checks, this makes absolutely sense, so please refactor this!

Reply:
I did split it in three methods and changed the corresponding calls from the front-end.
Previous endpoint also returned all complaints for current user which should be tutor without specifying courseId and exerciseId. I didn't find any usage for this, but still leave it as a separate GET `complaints` endpoint. Do we still need it?